src/runtime/analytics-service.js

```diff
@@ -192,22 +204,31 @@ export class AnalyticsService {
   }
 
   /**
-   * @param {!../proto/api_messages.AnalyticsEvent} event
+   * @param {!../api/client-event-manager-api.ClientEvent} event
    * @return {!AnalyticsRequest}
    */
   createLogRequest_(event) {
+    //ignore event.additionalParameters.  It may have data we shouldn't log
+    const meta = new AnalyticsEventMeta();
+    meta.setEventOriginator(event.eventOriginator);
+    meta.setIsFromUserAction(event.isFromUserAction);
+
     const request = new AnalyticsRequest();
-    request.setEvent(event);
+    request.setEvent(event.eventType);
     request.setContext(this.context_);
+    request.setMeta(meta);
     return request;
   }
 
   /**
    * @param {!../proto/api_messages.AnalyticsEvent} event
    */
   logEvent(event) {
-    this.lastAction_ = this.start_().then(port => {
-      port.message({'buf': this.createLogRequest_(event).toArray()});
+    this.eventManager_.logEvent({
+      eventType: event,
+      eventOriginator: EventOriginator.SWG_CLIENT,
+      isFromUserAction: null,
```

Comment:
Why not let internal events to provide isFromUserAction and additionalParams?
Make them optional so existing calls will pass.
/**
 * @param {!../proto/api_messages.AnalyticsEvent} event
 * @param {!boolean=} active
 * @param {!Object=} params
 */
logEvent(event, active, params) {
  const userAction = (active != null): active : null;
  this.eventManager_.logEvent({
    eventType: event,
    eventOriginator: EventOriginator.SWG_CLIENT,
    isFromUserAction: userAction,
    additionalParameters: params || null,
});

Reply:
I actually wasn't aware of the =} syntax for optional params - very helpful to know thank you.

I'm generally preferring to redirect things directly to events manager.  I was effectively planning to do that in a different set of changes to keep things limited in scope.  This just keeps things working the way they were before.  My thinking is that the analytics service is a secret background thing that most people don't need to be aware of.  It is just a listener and nothing should really need to interact with it directly.  Also they can just get eventManager from deps and eventually the analytics service can be removed as something publicly available in deps since things will need to know about it.  It will certainly still need to be instantiated and stored somewhere.

Reply:
I did improve the documentation for that and add the deprecated tag to make it clear.

Reply:
After digging into entitlements manager again: since it needs to directly call analytics service to set user state anyway I think it's appropriate to keep the helper log function here.  I added isFromUserAction as an optional parameter and removed the deprecated tag.  I think that in general the additionalParameters field existed for Propensity events and that I'd rather keep it out of the helper for now.  I could be convinced that it's useful though if you think its important.

Reply:
I think we can always add the additional Parameters if necessary. Its good enough to add the isFromUserAction though since the calling APIs know what event they are sending, they can set this value which is very useful.